build process for assets #55

UltraMC opened the pull request:
Change build process:
1. Work done on master branch with Tibia-11 changes merged into it
2. build.js file organised and changed:
- contents of new directory `/assets` are treated as template
- from it `/_build` directory contents are filled and files in there get hash replacement (as it was before this change)
- hash replacement in js file happens with new option which helps make js file in question readable and parseable
- instead of iterating over all directories, it's now configurable by whitelist
3. Introduced `/assets` directory - contains "local" files that are built to deploy-ready/work-on-local state. Contains addition of img, css, js and font files downloaded straight from windbot domain
4. All libs got organised into their respective lib group files
5. minor changes to index.html to load files from local assets instead of hotlinking to windbot domain

Commit:
build process for assets #55

Change build process:
1. Work done on master branch witch Tibia-11 changes merged into it
2. build.js file organised and changed:
- contents of new directory assets are treated as template
- from it _build directory contents are filled and files in there get hash replacement (as it was before this change)
- hash replacement in js file happens with new option which helps make js file in question readable and parseable
- instead of iterating over all directories, it's now configurable by whitelist
3. Introduced assets directory - contains "local" files that are built to deploy-ready/work-on-local state. Contains addition of img, css, js and font files downloaded straight from windbot domain
4. All libs got organised into their respective lib group files
5. minor changes to index.html to load files from local assets instead of hotlinking to windbot domain

rmobis reviewed (changes requested):
I'd really prefer if we kept each external library on its own file. This makes it easier to apply eventual updates and might also saves us from some headache when deploying, since every other page have them separate.

UltraMC commented on assets/docs.html:
```diff
@@ -150,33 +147,34 @@ <h2 class="docs-subtitle">Contributors</h2>
+						<a href="features.html">» Features</a><br /><br />
+						<a href="download.html">» Download</a><br /><br />
+						<a href="purchase.html">» Purchase</a><br /><br />
+						<a href="lua-variables.html">» Documentation</a><br /><br />
```
Long story short its dependent on doctype, mostly optional as stated in specs, in practice required for proper syntax highlighting. Anyway it's good to have tags always closing. 

Ultimately it's a matter of code style which we don't have. I'm not the one accepting PR's.

UltraMC replied:
@ lib files

OK I'll introduce PR with them separately. I was afraid that you'll rice concerns about that one but for security point of view. In theory it's hard to verify lib files (I could put there whatever I want minified).

You say it's better for maintaining I say final goal would be to get rid of them completely in favor for dependencies manager.

Other solution is to rely on official CDNs.

Agains that is that one big file downloads faster than multiple small files and either way we should have concater+minifier.

Can you reflect on that?

rmobis replied:
I agree the ideal goal is to have them managed by npm and have some automated concatenation and minification, not only for performance reasons but also for security and ease of use. 

However, that would mean that, in order to make sure we're using the same files on the whole website, we'd have to host WindBot's entire website on a repository and use the same build system we think of here, and I just don't think we're ready to start messing with any other part of the website right now.

*PS:* I'm sorry if my comment about the `<br />` tags sounded condescending. It's just that you implied to be new on open source contributions and I thought I'd give you a heads up.

Commit:
build process for assets #55

Resigned from using locally hosted assets in favor of hotlinking to live original for:
- it's bad practise to store libs in repo
- bundling lib files together would require change of other pages of website which this repo is out of scope

In result introduced new _build file - docs_local.html

rmobis commented:
PR merged; thanks
